pkg/apiutil/serverapi/middleware.go

```diff
@@ -40,32 +41,76 @@ const (
 
 var initHTTPClientOnce sync.Once
 
+// dialClient used to dail http request.
+var dialClient = &http.Client{
+	Transport: &http.Transport{
+		DisableKeepAlives: true,
+	},
+}
+
+type runtimeServiceAuth struct {
+	s *server.Server
+}
+
+// NewRuntimeServiceAuth checks if the path is invalid.
+func NewRuntimeServiceAuth(s *server.Server) *runtimeServiceAuth {
```

Comment:
```suggestion
func NewRuntimeServiceAuth(s *server.Server) Handler {
```
Exported function with unexported return type